### DISCUSSION TITLE
Payload License is (still) not working

### DISCUSSION THREAD

**Discussion by Duckinm** [payloadcms/payload#299] [General]
I already send the information to dev@payloadcms.com yesterday, but yeah, I need a fast response. Have to stick to the deadline and future works. so I decided to post it here too.

My Payload is hosting at Heroku which already set the env config for a license. More than 20 times of validation, everything seems correct but it still said that my license is invalid.

<img width="1440" alt="Screen Shot 2564-08-15 at 12 54 26 AM" src="https://user-images.githubusercontent.com/29366271/129469545-77a0945c-6cc4-47cd-9431-81fce5be64fd.png">

As I open-sourced my project, I will share it here.
https://github.com/Duckinm/merchance-asiana

Looking forward to hearing your response!
Thank you.

**Comment by richardvanbergen** [comment]
Hey I'm no payload dev but is your domain asainresidence.com listed here?

![image](https://user-images.githubusercontent.com/3854592/129472368-22b7e8a3-bd3c-4224-b076-8851e9b7c114.png)

And... I know this sounds obvious but it confused me too in the rush of trying to put things online. Have you tried loading just `/admin`? That page (`/admin/unlicenced-domain`) doesn't detect changes in licenses. The main admin route does.

I've gone through this process and it did seem to pick up the license changes almost immediately in my experience.

**Comment by Duckinm** [comment]
I already tried everything you’ve mentioned.. I also use every browser in my machine.

**Comment by denolfe** [comment]
@Duckinm We're receiving your license as `undefined`. I've noticed your `dist/index.js` has `process.env.PAYLOAD_LICENSE` instead of `process.env.PAYLOAD_LICENSE_KEY` like your `.env` file specifies. However, your .ts code looks correct, you may need to just recompile.

TS code (correct): https://github.com/Duckinm/merchance-asiana/blob/main/server/index.ts#L12
Compiled code (incorrect): https://github.com/Duckinm/merchance-asiana/blob/main/dist/index.js#L52

This mismatch looks to be why it's being sent as `undefined`

**Reply by denolfe** [reply]
Confirmed this was resolved. License is now being received properly.
